### PR TITLE
Pimcore 11 Support

### PR DESCRIPTION
**Pull request by fashxp**
- added Pimcore 11 support
- switched to new bundle structure #9
- dropped Pimcore 10 support

### CODE REVIEW

**Comment by martineiber** [comment]
I have created a 2.x branch, and we should merge it into this branch, right?

**Comment by dvesh3** [comment]
@martineiber we can merge this to 1.x and rename 1.x to 2.x since we don't have any pending minor changes to tag on the branch.

File: src/Resources/skeleton/bundle/Bundle.php.twig [hunk 1]
```diff
@@ -2,17 +2,39 @@
 
 {% block use_statements %}
 use Pimcore\Extension\Bundle\AbstractPimcoreBundle;
+use Pimcore\Bundle\AdminBundle\Support\PimcoreBundleAdminSupportInterface;
```
Comment (martineiber) [inline src/Resources/skeleton/bundle/Bundle.php.twig]:
```suggestion
use Pimcore\Extension\Bundle\PimcoreBundleAdminClassicInterface;
use Pimcore\Extension\Bundle\Traits\BundleAdminClassicTrait;
```

File: src/Resources/skeleton/bundle/Bundle.php.twig [hunk 2]
```diff
@@ -2,17 +2,39 @@
...
 {% endblock use_statements %}
 
 {% block class_definition %}
-class {{ bundle }} extends AbstractPimcoreBundle
+class {{ bundle }} extends AbstractPimcoreBundle implements PimcoreBundleAdminSupportInterface
```
Comment (martineiber) [inline src/Resources/skeleton/bundle/Bundle.php.twig]:
```suggestion
class {{ bundle }} extends AbstractPimcoreBundle implements PimcoreBundleAdminClassicInterface
```

File: src/Resources/skeleton/bundle/Bundle.php.twig [hunk 3]
```diff
@@ -2,17 +2,39 @@
...
 {% endblock class_definition %}
 
 {% block class_body %}
-    public function getJsPaths()
+    public function getPath(): string
```
Comment (martineiber) [inline src/Resources/skeleton/bundle/Bundle.php.twig:16]:
```suggestion
    use BundleAdminClassicTrait;

    public function getPath(): string
```

**Review by martineiber — CHANGES_REQUESTED** [review]
Switch to PimcoreBundleAdminClassicInterface.

File: src/Resources/skeleton/bundle/Bundle.php.twig [hunk 4]
```diff
@@ -2,17 +2,39 @@
...
+    public function getEditmodeJsPaths(): array
+    {
+        return [];
+    }
+
+    public function getEditmodeCssPaths(): array
+    {
+        return [];
+    }
+
```
Comment (dvesh3) [inline src/Resources/skeleton/bundle/Bundle.php.twig]:
```suggestion
```
not required when using `BundleAdminClassicTrait`

File: README.md [hunk 5]
```diff
@@ -10,16 +10,20 @@ follow the instructions:
...
 
-On your Pimcore X root project
+1. On your Pimcore 11 root project
 
 ```bash
 $ composer require pimcore/bundle-generator
 ```
 
-After that you should enable it using Pimcore Extension Manager on Admin or by using this command
+2. Make sure the bundle is enabled in the `config/bundles.php` fiel. The following lines should be added:
```
Comment (dvesh3) [inline README.md]:
```suggestion
2. Make sure the bundle is enabled in the `config/bundles.php` file. The following lines should be added:
```

**Comment by fashxp** [comment]
@dvesh3 @martineiber please apply all the changes and retest then the generated bundles, thx.

**Review by martineiber — APPROVED** [review]
LGTM

**Review by dvesh3 — APPROVED** [review]

**Pull request merged**